### ISSUE TITLE
Loklak Docker Container error: git not found + Could not initialize DAO + crash

### ISSUE THREAD

**Issue by niccokunzmann** [loklak/loklak_server#1526]
- Issue type: Bug report 

### Short description

Loklak docker container fails to start

### Environment

 - Operating system: ubuntu, docker
 - Software version: tag latest
 - Software source: dockerhub mariobehling:loklak 

### Steps to reproduce

1. 
2. 
3. 
pull the container
run the container
These are the logs from several start attempts:
```
loklak_1                | 2017-09-09 02:12:01.625:INFO :NodeEnvironment:main: [Red Skull II] heap size [989.8mb], compressed ordinary object pointers [true]
loklak_1                | 2017-09-09 02:12:02.391:WARN :LoklakServer:main: null
loklak_1                | 2017-09-09 02:12:02.404:WARN :LoklakServer:main: Could not initialize DAO. Exiting.
loklak_1                | /loklak_server/bin/start.sh: line 9: git: command not found
loklak_1                | /loklak_server/bin/start.sh: line 10: git: command not found
loklak_1                | starting loklak
loklak_1                | 2017-09-10 02:03:12.571:INFO :Log:main: Logging initialized @8565ms
loklak_1                | 2017-09-10 02:03:12.590:INFO :LoklakServer:main: Starting loklak initialization
loklak_1                | 2017-09-10 02:03:13.467:INFO :DAO:main: initializing loklak DAO
loklak_1                | 2017-09-10 02:03:13.701:INFO :DAO:main: Key pair loaded from file. Peer hash: SHA-256 gexC6SHJzT1ZzBVlNLXlREE74NB5ILkVfCqDZkAentY=
loklak_1                | 2017-09-10 02:03:15.534:INFO :Node:main: [Puma] version[2.4.0], pid[1], build[ce9f0c7/2016-08-29T09:14:17Z]
loklak_1                | 2017-09-10 02:03:15.534:INFO :Node:main: [Puma] initializing ...
loklak_1                | 2017-09-10 02:03:15.574:INFO :PluginsService:main: [Puma] modules [], plugins [], sites []
loklak_1                | 2017-09-10 02:03:16.686:INFO :NodeEnvironment:main: [Puma] using [1] data paths, mounts [[/loklak_server/data (/dev/mapper/kleinerweisser--vg-root)]], net usable_space [1.4tb], net total_space [1.7tb], spins? [possibly], types [ext4]
loklak_1                | 2017-09-10 02:03:16.692:INFO :NodeEnvironment:main: [Puma] heap size [989.8mb], compressed ordinary object pointers [true]
loklak_1                | 2017-09-10 02:03:16.862:WARN :LoklakServer:main: null
loklak_1                | 2017-09-10 02:03:16.863:WARN :LoklakServer:main: Could not initialize DAO. Exiting.
loklak_1                | /loklak_server/bin/start.sh: line 9: git: command not found
loklak_1                | /loklak_server/bin/start.sh: line 10: git: command not found
loklak_1                | starting loklak
loklak_1                | 2017-09-10 11:09:05.397:INFO :Log:main: Logging initialized @5923ms
loklak_1                | 2017-09-10 11:09:05.407:INFO :LoklakServer:main: Starting loklak initialization
loklak_1                | 2017-09-10 11:09:06.339:INFO :DAO:main: initializing loklak DAO
loklak_1                | 2017-09-10 11:09:06.576:INFO :DAO:main: Key pair loaded from file. Peer hash: SHA-256 gexC6SHJzT1ZzBVlNLXlREE74NB5ILkVfCqDZkAentY=
loklak_1                | 2017-09-10 11:09:08.816:INFO :Node:main: [Monstra] version[2.4.0], pid[1], build[ce9f0c7/2016-08-29T09:14:17Z]
loklak_1                | 2017-09-10 11:09:08.819:INFO :Node:main: [Monstra] initializing ...
loklak_1                | 2017-09-10 11:09:08.863:INFO :PluginsService:main: [Monstra] modules [], plugins [], sites []
loklak_1                | 2017-09-10 11:09:09.107:INFO :NodeEnvironment:main: [Monstra] using [1] data paths, mounts [[/loklak_server/data (/dev/mapper/kleinerweisser--vg-root)]], net usable_space [1.4tb], net total_space [1.7tb], spins? [possibly], types [ext4]
loklak_1                | 2017-09-10 11:09:09.111:INFO :NodeEnvironment:main: [Monstra] heap size [989.8mb], compressed ordinary object pointers [true]
loklak_1                | 2017-09-10 11:09:09.240:WARN :LoklakServer:main: null
loklak_1                | 2017-09-10 11:09:09.241:WARN :LoklakServer:main: Could not initialize DAO. Exiting.
```

### Expected behaviour

I would like loklak to run in the container.

### Actual behaviour

It does not start.

### Usecase

I would like to host loklak in a docker container.

**Comment by singhpratyush** [comment]
Hi, @niccokunzmann!

Please use `loklak/loklak_server` image instead. I am able to run loklak using it - 

```bash
docker run -p 9000:80 loklak/loklak_server
```

**Comment by niccokunzmann** [comment]
Thanks for your reply. I tried the following:


```
startup-job@quelltext:~/server/services$ docker-compose create
Pulling loklak (loklak/loklak_server:latest)...
latest: Pulling from loklak/loklak_server
88286f41530e: Already exists
ea62e6e7acd3: Pull complete
44b9ab95ad04: Pull complete
88dec8c39c93: Pull complete
d07c323d4fd0: Pull complete
3ff5328d9867: Pull complete
795f6f81e359: Pull complete
39f545cebf3d: Pull complete
0e83fbb1b3f7: Pull complete
58ace5282cda: Pull complete
4449b43f9929: Pull complete
5feb69a87f1c: Pull complete
Digest: sha256:7ef2128b090dbe741c208e0053ec69ed39cd2f89247e339a59258db7b25b9338
Status: Downloaded newer image for loklak/loklak_server:latest
startup-job@quelltext:~/server/services$ docker-compose rm loklak
Going to remove services_loklak_1
Are you sure? [yN] y
Removing services_loklak_1 ... done
startup-job@quelltext:~/server/services$ docker-compose create loklak
Creating services_loklak_1
startup-job@quelltext:~/server/services$ docker-compose logs lokalk
ERROR: Aborting.
startup-job@quelltext:~/server/services$ docker-compose logs loklak
Attaching to services_loklak_1
startup-job@quelltext:~/server/services$ docker-compose start loklak
Starting loklak ... done
startup-job@quelltext:~/server/services$ docker-compose logs loklak
Attaching to services_loklak_1
loklak_1                | /loklak_server/bin/start.sh: line 9: git: command not found
loklak_1                | /loklak_server/bin/start.sh: line 10: git: command not found
loklak_1                | starting loklak
startup-job@quelltext:~/server/services$ docker-compose ps
             Name                            Command                State                      Ports
----------------------------------------------------------------------------------------------------------------------
services_loklak_1                 /loklak_server/bin/start.s ...   Exit 255

```

My `docker-compose.yml`:

```
  loklak:
    image: loklak/loklak_server:latest
    networks:
    - proxy
    volumes:
    - "/srv/docker/loklak/data:/loklak_server/data"
    cpuset: "1"

```

**Comment by niccokunzmann** [comment]
These lines cause an error output: https://github.com/loklak/loklak_server/blob/development/bin/start.sh#L10
I believe, git is not installed. Maybe that is a problem. I can not speak for the " Could not initialize DAO" message.

**Comment by singhpratyush** [comment]
@niccokunzmann: That line is for getting latest git commit details and saving them as env vars. The git directory is not added to the Docker image hence the command fails.

Though I don't know why it works with `docker run` and not with `docker-compose`. I'll send a pull request resolving the issue.

**Comment by mariobehling** [comment]
@niccokunzmann @singhpratyush Is this issue still there?

**Comment by singhpratyush** [comment]
@mariobehling: Fixing this now.

**Comment by singhpratyush** [comment]
Fixed in 6eb6dd7f7416f49318159a036648126793617b0d.
